spec/lib/api/2.0/views-spec.js

```diff
@@ -0,0 +1,147 @@
+    describe('2.0 Views', function() {
+        // should create a view (octet-stream)
+        it('should get all views', function() {
+            var views = [
+                {name: 'foo', content: 'foo', scope: 'global'},
+                {name: 'bar', content: 'bar', scope: 'global'}
+            ];
+            viewsProtocol.getAll.resolves(views);
+            return helper.request().get('/api/2.0/views')
+                .expect('Content-Type', /^application\/json/)
+                .expect(200)
+                .expect(function(res) {
+                    _(res.body).forEach(function(view, i) {
+                        expect(view).to.deepEqual(views[i]);
+                    });
+                    expect(viewsProtocol.getAll).to.have.been.calledOnce;
+                });
+        });
+
+        it('should get one view', function() {
+            var view = {name: 'foo', content: 'foo', scope: 'global'};
+
+            viewsProtocol.get.resolves(view);
+            return helper.request().get('/api/2.0/views/foo')
+                .expect('Content-Type', /^application\/json/)
+                .expect(200)
+                .expect(function(res) {
+                    expect(res.body).to.deep.equal(view);
+                    expect(viewsProtocol.get).to.have.been.calledOnce;
+                    expect(viewsProtocol.get).to.have.been.calledWith('foo');
+                });
+        });
+
+        it('should fail to get non-existant view', function() {
+            viewsProtocol.get.resolves();
+            return helper.request().get('/api/2.0/views/foo')
+                .expect('Content-Type', /^application\/json/)
+                .expect(404)
+                .expect(function() {
+                    expect(viewsProtocol.get).to.have.been.calledOnce;
+                    expect(viewsProtocol.get).to.have.been.calledWith('foo');
+                });
+        });
+
+        it('should create a text/plain view', function() {
+            var view = {name: 'foo', content: 'foo', scope: 'global'};
+
+            viewsProtocol.put.resolves(view);
+            return helper.request().put('/api/2.0/views/foo')
+                .set('Content-Type', 'text/plain')
+                .send('{ "message": "hello" }')
+                .expect('Content-Type', /^application\/json/)
+                .expect(200)
+                .expect(function(res) {
+                    expect(res.body).to.deep.equal(view);
+                    expect(viewsProtocol.put).to.have.been.calledOnce;
+                });
+        });
+
+        it('should create an application/octet-stream view', function() {
+            var view = {name: 'foo', content: 'foo', scope: 'global'};
+            var req = helper.request().put('/api/2.0/views/foo');
+
+            viewsProtocol.put.resolves(view);
+            req.set('Content-Type', 'application/octet-stream');
+            req.write(new Buffer('{ "message": "hello" }', 'ascii'));
+            req.end(function(err, res) {
```

Comment:
I know less about this behavior, but this looks like it gets called asynchronously? In which case the test won't actually catch these errors?